File: mobile-ios/App/Sources/App/AppLanguage.swift
import Foundation

/// Backend içeriğinin (ürün adı, kategori, duyuru…) hangi dilde isteneceği.
/// Tema gibi YEREL bir tercih — profile veya cihaz diline bağlı değil, cihazda kalır.
/// Varsayılan Türkçe. `UserDefaults` anahtarı `SettingsView`'daki `@AppStorage`
/// ile aynı ("contentLanguage").
enum AppLanguage {
    static let supported = ["tr", "en", "de", "fr", "ar", "nl"]
    static let fallback = "tr"
    static let storageKey = "contentLanguage"

    /// Üretimde `.standard` (SwiftUI `@AppStorage` ile aynı depo). Yalnızca testler
    /// geçici bir suite ile değiştirir — bu yüzden `nonisolated(unsafe)`.
    nonisolated(unsafe) static var store: UserDefaults = .standard

    static var current: String {
        let stored = store.string(forKey: storageKey)
        if let stored, supported.contains(stored) { return stored }
        return fallback
    }

    static func setCurrent(_ code: String) {
        guard supported.contains(code) else { return }
        store.set(code, forKey: storageKey)
    }

    /// Ayarlar ekranında gösterilecek yerel dil adı.
    static func displayName(_ code: String) -> String {
        switch code {
        case "tr": "Türkçe"
        case "en": "English"
        case "de": "Deutsch"
        case "fr": "Français"
        case "ar": "العربية"
        case "nl": "Nederlands"
        default: code
        }
    }
}

File: mobile-android/app/src/main/java/com/erenlermarket/app/data/AppLanguage.kt
package com.erenlermarket.app.data

/**
 * Backend içeriğinin (ürün adı, kategori, duyuru…) hangi dilde isteneceği.
 * Tema gibi YEREL bir tercih — profile bağlı değil, cihazda kalır.
 * Gerçek kaynak `LanguageController` (DataStore); burası sadece ağ
 * interceptor'ının DI'sız okuyabilmesi için tutulan process içi kopya.
 * Varsayılan Türkçe.
 */
object AppLanguage {
    /** Backend'in desteklediği diller (yerel adlarıyla). */
    val supported = listOf("tr", "en", "de", "fr", "ar", "nl")

    const val DEFAULT = "tr"

    @Volatile
    var current: String = DEFAULT
        internal set

    /** Ayarlar ekranında gösterilecek yerel dil adı. */
    fun displayName(code: String): String = when (code) {
        "tr" -> "Türkçe"
        "en" -> "English"
        "de" -> "Deutsch"
        "fr" -> "Français"
        "ar" -> "العربية"
        "nl" -> "Nederlands"
        else -> code
    }
}
